spec/models/mixins/event_mixin_spec.rb

```diff
@@ -68,4 +68,273 @@ def event_where_clause(assoc)
       expect(test_class.new).not_to have_events
     end
   end
+
+  context "event_where_clause" do
+    let(:ems_id)       { 1234 }
+    let(:project_name) { "container_project_name" }
+    let(:known_missing_event_mixin_classes) { [MiqServer, Storage, ResourcePool, ContainerImage] }
+
+    # TODO: The classes: Storage, ContainerImage, ResourcePool:
+    #  * Do NOT have _id columns in event_streams
+    #  * Do NOT include EventMixin
+    #  * Are included in MiqEvent::SUPPORTED_POLICY_AND_ALERT_CLASSES
+    #
+    # TODO: The classes:  AvailabilityZone, Container, PhysicalChassis, PhysicalSwitch
+    #   * Do have id columns in event_streams
+    #   * Do include EventMixin
+    #   * Are NOT included in MiqEvent::SUPPORTED_POLICY_AND_ALERT_CLASSES
+    context "AvailabilityZone" do
```

Comment:
Wondering if this is clearer as a shared_examples, which each of the models verify.  The ones that have overriding behavior will demonstrate that in their own tests.